snowflake.go

```diff
@@ -0,0 +1,328 @@
+		}
+	}
+
+	if req.Expiration != nil {
+		err = s.updateUserExpiration(ctx, tx, req.Username, req.Expiration)
+		if err != nil {
+			return dbplugin.UpdateUserResponse{}, err
+		}
+	}
+
+	if err := tx.Commit(); err != nil {
+		return dbplugin.UpdateUserResponse{}, err
+	}
+
+	return dbplugin.UpdateUserResponse{}, nil
+}
+
+func (s *SnowflakeSQL) updateUserPassword(ctx context.Context, tx *sql.Tx, username string, req *dbplugin.ChangePassword) error {
+	password := req.NewPassword
+
+	if username == "" || password == "" {
+		return fmt.Errorf("must provide both username and password to modify password")
+	}
+
+	stmts := req.Statements.Commands
+	if len(stmts) == 0 {
+		stmts = []string{defaultSnowflakeRotateCredsSQL}
+	}
+
+	for _, stmt := range stmts {
+		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+			query = strings.TrimSpace(query)
+			if len(query) == 0 {
+				continue
+			}
+
+			m := map[string]string{
+				"name":     username,
+				"username": username,
+				"password": password,
+			}
+
+			if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
+				return fmt.Errorf("failed to execute query: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (s *SnowflakeSQL) updateUserExpiration(ctx context.Context, tx *sql.Tx, username string, req *dbplugin.ChangeExpiration) error {
+	expiration := req.NewExpiration
+
+	if username == "" || expiration.IsZero() {
+		return fmt.Errorf("must provide both username and valid expiration to modify expiration")
+	}
+
+	expirationStr, err := calculateExpirationString(expiration)
+	if err != nil {
+		return err
+	}
+
+	stmts := req.Statements.Commands
+	if len(stmts) == 0 {
+		stmts = []string{defaultSnowflakeRenewSQL}
+	}
+
+	for _, stmt := range stmts {
+		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+			query = strings.TrimSpace(query)
+			if len(query) == 0 {
+				continue
+			}
+
+			m := map[string]string{
+				"name":       username,
+				"username":   username,
+				"expiration": expirationStr,
+			}
+
+			if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
+				return fmt.Errorf("failed to execute query: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (s *SnowflakeSQL) DeleteUser(ctx context.Context, req dbplugin.DeleteUserRequest) (dbplugin.DeleteUserResponse, error) {
+	s.Lock()
+	defer s.Unlock()
+
+	username := req.Username
+	statements := req.Statements.Commands
+	if len(statements) == 0 {
+		statements = []string{defaultSnowflakeDeleteSQL}
+	}
+
+	db, err := s.getConnection(ctx)
+	if err != nil {
+		return dbplugin.DeleteUserResponse{}, err
+	}
+
+	for _, stmt := range statements {
+		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+			query = strings.TrimSpace(query)
+			if len(query) == 0 {
+				continue
+			}
+
+			m := map[string]string{
+				"name":     username,
+				"username": username,
+			}
+			if err := dbtxn.ExecuteDBQuery(ctx, db, m, query); err != nil {
```

Comment:
Similar to `NewUser()`, wondering if each of these statements being executed should be in a transaction? Similar to [postgresql.go#L311](https://github.com/hashicorp/vault/blob/master/plugins/database/postgresql/postgresql.go#L311).